src/zenml/stack/utils.py

```diff
@@ -139,3 +140,33 @@ def get_flavor_by_name_and_type_from_zen_store(
             f"'{component_type}' exists."
         )
     return flavors[0]
+
+
+class stack_context:
+    """Context handler to reset the original active stack."""
+
+    def __init__(self) -> None:
+        """Constructor for stack_context saves active stack."""
+        self._default_stack = Client().active_stack
+
+    def __enter__(self) -> None:
+        """Enters in the stack context."""
+        pass
+
+    def __exit__(
+        self,
+        exception_type: Optional[Type[BaseException]],
+        exception_value: Optional[BaseException],
+        traceback: Optional[TracebackType],
+    ) -> None:
+        """Get a stack component flavor by name and type from a ZenStore.
+
+        Args:
+            exception_type: Type of the exception that was raised.
+                None if no exception.
+            exception_value: Type of exception that was raised.
+                e.g., divide_by_zero error. None if no exception.
+            traceback: Traceback report. None if no exception.
+        """
+        if self._default_stack.id != Client().active_stack:
```

Comment:
There is a small error here. This `if` condition compares a UUID to an actual `Stack` object, thus it will always return `False`.